src/lib/y2storage/proposal/devicegraph_generator.rb

```diff
@@ -155,6 +155,7 @@ def refine_planned_partitions!(planned_partitions, deleted_partitions)
         new_swap_volumes.each_with_index do |swap_volume, idx|
           deleted_swap = deleted_swaps[idx]
           break unless deleted_swap
+          break unless deleted_swap.formatted_as?(:swap)
```

Comment:
This will raise an exception if `new_swap_volumes.size > deleted_swaps.size` (very common scenario, I would say).

Also in this scenario, this change will produce the UUID to NOT be reused:
```ruby
planned_partitions =  [a_swap_without_reuse]
deleted_partitions = [an_ext4_with_wrong_id_of_swap, a_real_swap]
```

A more robust fix (and more understandable IMO) would be to change the first line like this:
```ruby
deleted_swaps = deleted_partitions.select { |part| part.id.is?(:swap) && part.formatted_as?(:swap) }
```

Isn't it?